### DISCUSSION TITLE
How do i make a TXT record

### DISCUSSION THREAD

**Discussion by Tab36** [community/community#143425] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area
https://github.com/orgs/community/discussions
Question

### Body

I am a newcomer to github and i want to get a custom domain. I read all the docs and did what it said until I had to make a TXT record. I don't know what that means or how to do it, so it would be nice if one of you could explain it to me.

**THIS IS ANSWERED - I FOUND OUT THAT I NEDDED TO BUY DOMAINS EXTERNALLY. THANKS TO @Aiche-H FOR HIS/HER DETAILED ANSWER!**

**Comment by Aiche-H** [comment, marked as answer]
**Understanding TXT Records**

A TXT record, short for Text Record, is a type of DNS record that stores simple text strings. In the context of verifying a custom domain with GitHub Pages, it's used to prove that you own the domain.

**How to Create a TXT Record**

1. **Access Your Domain Registrar:**
   - This is the company where you purchased your domain name (e.g., GoDaddy, Namecheap, Google Domains).
   - Log in to your account.

2. **Navigate to DNS Settings:**
   - Look for a section labeled "DNS," "Domain Names," or something similar.
   - This might involve going to your domain's control panel or a specific DNS management tool.

3. **Add a New TXT Record:**
   - You'll typically see options to add different types of records, including TXT.
   - Click on "Add Record" or a similar button.
   - Select "TXT" as the record type.

4. **Fill in the Required Fields:**
   - **Host/Name:** This field will vary depending on your domain registrar. It might be empty, or you might need to enter a specific value (often "@").
   - **Points to/Value:** Paste the exact text provided by GitHub into this field. It will look something like this: `github-pages-verification-code=your_verification_code`
   - **TTL (Time to Live):** This determines how long the DNS record should be cached. You can usually leave the default value.

**Reply by Tab36** [reply]
So... I have to buy a domain externally?

**Reply by Aiche-H** [reply]
Yes, you need to purchase a domain name before you can use it for your custom domain.

**Reply by Tab36** [reply]
Oh. I thought I could set up a custom domain inside github. I guess I have buy one anyway. Thanks for your answer!

**Reply by AvaCaine** [reply]
what if i want to use a github.io page
